tests/unit/test_solvers.py

```diff
@@ -202,6 +209,11 @@ def test_functions(x):
             def test_functions(x):
                 norm_x = jnp.linalg.norm(x)
                 return jnp.array([norm_x, 2 * norm_x, 2 + norm_x])
+        elif request.param == "excessive_test_functions":
+
+            def test_functions(x):
+                del x
+                return jnp.zeros((len(nodes) + 1,))
```

Comment:
I am not sure what `excessive_test_functions` is trying to do, and why `test_functions` looks the way it looks? (I am assuming we are trying to explore what happens when there are more `test_functions` than there are datapoints, and we are just returning zeros as they are the simplest functions?)